Userland/Libraries/LibJS/Runtime/AsyncFromSyncIteratorPrototype.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2021, David Tuin <davidot@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/AsyncFromSyncIteratorPrototype.h>
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/IteratorOperations.h>
+#include <LibJS/Runtime/Promise.h>
+#include <LibJS/Runtime/PromiseConstructor.h>
+#include <LibJS/Runtime/PromiseReaction.h>
+
+namespace JS {
+
+AsyncFromSyncIteratorPrototype::AsyncFromSyncIteratorPrototype(GlobalObject& global_object)
+    : PrototypeObject(*global_object.async_iterator_prototype())
+{
+}
+
+void AsyncFromSyncIteratorPrototype::initialize(GlobalObject& global_object)
+{
+    auto& vm = global_object.vm();
+    Object::initialize(global_object);
+
+    u8 attr = Attribute::Writable | Attribute::Configurable;
+    define_native_function(vm.names.next, next, 1, attr);
+    define_native_function(vm.names.return_, return_, 1, attr);
+    define_native_function(vm.names.throw_, throw_, 1, attr);
+}
+
+AsyncFromSyncIteratorPrototype::~AsyncFromSyncIteratorPrototype()
+{
+}
+
+// 27.1.4.4 AsyncFromSyncIteratorContinuation ( result, promiseCapability ), https://tc39.es/ecma262/#sec-asyncfromsynciteratorcontinuation
+static Object* async_from_sync_iterator_continuation(VM& vm, GlobalObject& global_object, Object& result, PromiseCapability& promise_capability)
```

Comment:
I don't know why it is being ignored in the current version of `TRY_OR_REJECT`, but `IfAbruptRejectPromise` contains `? Call()`, so I think we need to fix the macro and then this must return `ThrowCompletionOr`.

Reply:
All the callers do so with `!` but I made it return a TCO and called with `MUST` for now.